datafusion/sql/src/unparser/rewrite.rs

```diff
@@ -363,3 +363,138 @@ impl TreeNodeRewriter for TableAliasRewriter<'_> {
         }
     }
 }
+
+/// Takes an input list of identifiers and a list of identifiers that are available from relations or joins.
+/// Removes any table identifiers that are not present in the list of available identifiers, retains original column names.
+pub fn remove_dangling_identifiers(idents: &mut Vec<Ident>, available_idents: &[String]) {
```

Comment:
I don't understand this code super deeply, but this seems to me like it is treating the symptom (incorrect qualifiers) rather than the root cause. 

Specifically, did you look into fixing the code so that it didn't create incorrect indentifiers in the first place, rather than trying to modify the created AST after the fact to remove incorrect indentifers ?

Reply:
Yeah, I had taken a look into doing this at the unparser `LogicalPlan` level but I wasn't making very good progress. It could be my lack of understanding with `LogicalPlan`, but I think the symptom originates from the parser rather than the unparser.

If you'd be open to merging this still as an AST modifier, perhaps we could gate it behind a dialect option or feature flag as a non-default?